components/eam/src/chemistry/aerosol/wetdep.F90

```diff
@@ -502,798 +471,732 @@ subroutine wetdepa_v2( ncol, deltat, &
+   else
+        if (precabx_base_old < prec_smallaa) then
+            resusp_x = 0.0_r8
+        else
+            ! non-linear resuspension of aerosol number based on raindrop number
+            u_old = min_max_bound(0.0_r8, 1.0_r8, precabx_old/precabx_base_old)
+            x_old = 1.0_r8 - fprecn_resusp_vs_fprec_evap_mpln(1.0_r8-u_old, is_st_cu)
+            x_old = min_max_bound(0.0_r8, 1.0_r8, x_old)
+            x_new = 0.0_r8
+            resusp_x = max( 0.0_r8, precnumx_base_old*(x_old - x_new) )
+        endif
+   endif
+   ! setting both these precip rates to zero causes the resuspension
+   ! calculations to start fresh if there is any more precip production
+   precabx_new = 0.0_r8
+   precabx_base_new = 0.0_r8
+
+   end subroutine wetdep_resusp_noprecip
+
+!==============================================================================
+   subroutine wetdep_resusp_nonlinear(                          &
+                        is_st_cu,       mam_prevap_resusp_optcc,& ! in
+                        precabx_old,    precabx_base_old,       & ! in
+                        scavabx_old,    precnumx_base_old,      & ! in
+                        precabx_new,                            & ! in
+                        scavabx_new,    resusp_x                ) ! out
+
+! ------------------------------------------------------------------------------
+! do nonlinear resuspension of aerosol mass or number
+! ------------------------------------------------------------------------------
+   integer, intent(in) :: is_st_cu      ! options for stratiform (1) or convective (2) clouds
+                                        ! raindrop size distribution is
+                                        ! different for different cloud:
+                                        ! 1: assume marshall-palmer distribution
+                                        ! 2: assume log-normal distribution
+   integer, intent(in) :: mam_prevap_resusp_optcc       ! suspension options
+   real(r8),intent(in) :: precabx_base_old ! input of precipitation at cloud base [kg/m2/s]
+   real(r8),intent(in) :: precabx_old  ! input of precipitation above this layer [kg/m2/s]
+   real(r8),intent(in) :: scavabx_old  ! input scavenged tracer flux from above [kg/m2/s]
+   real(r8),intent(in) :: precnumx_base_old ! precipitation number at cloud base [#/m2/s]
+   real(r8),intent(in) :: precabx_new  ! output of precipitation above this layer [kg/m2/s]
+   real(r8),intent(out) :: scavabx_new ! output scavenged tracer flux from above [kg/m2/s]
+   real(r8),intent(out) :: resusp_x    ! aerosol mass re-suspension in a particular layer [kg/m2/s]
+
+
+   ! local variables
+   real(r8)     :: u_old,u_new  ! fraction of precabx and precabx_base
+   real(r8)     :: x_old,x_new  ! fraction after calling function *_resusp_vs_fprec_evap_mpln
+   real(r8)     :: x_ratio      ! fraction of x_tmp/x_old
+   real(r8), parameter :: x_smallaa = 1.0e-30_r8     ! small value for x_old
```

Comment:
We have a lot of these "small values" with similar values. Is it worth considering whether to consolidate some of them?

Reply:
I agree, we should collect them all and declare at one place so that we can understand the differences and consolidate if they are repeated.